napari/layers/image/image.py

```diff
@@ -1016,7 +1051,8 @@ def _get_state(self):
                 'multiscale': self.multiscale,
                 'colormap': self.colormap.name,
                 'contrast_limits': self.contrast_limits,
-                'interpolation': self.interpolation,
```

Comment:
This will break plugins that used the `interpolation` key when receiving a layer data tuple. My guess is that's not many plugins (if any at all) and the fix is pretty simple, so I'm OK with removing this, but just wanted to flag.

Reply:
fix is not so easy as dis dict need to match constructor arguments. need to pop `interpolation` before pass to constructor is why I do not add it. 

but maybe @tlambert03 has other opinion/experience?

Reply:
Yeah, this is a recurring issue with deprecations :/

Reply:
Yeah, I'm aware of that problem and it irks me. To clarify, I definitely don't expect a fix in this PR and I think it's OK to break things here.